src/oemof/solph/_plumbing.py

```diff
@@ -73,21 +111,61 @@ class _Sequence(UserList):
     >>> s[8]
     42
 
+    >>> s = _Sequence(period_values=[0,1,2], highest_index=9)
+    >>> len(s)
+    9
+    >>> s[1]
+    0
+    >>> s[4]
+    1
+    >>> s[7]
+    2
+    >>> s
+    [0, 0, 0, 1, 1, 1, 2, 2, 2]
+
 
     """
 
-    def __init__(self, *args, **kwargs):
-        self.default = kwargs["default"]
+    def __init__(
+        self,
+        *args,
+        default=None,
+        period_values=None,
+        highest_index=-1,
+        **kwargs,
+    ):
+        if all([default is not None, period_values is not None]):
+            raise ValueError("Only default or periods must be given.")
+        self.default = default
+        self.period_values = period_values
         self.default_changed = False
-        self.highest_index = -1
+        self.highest_index = highest_index
+        self.period_length = (
+            int(highest_index / len(period_values)) if period_values else None
+        )
         super().__init__(*args)
 
+    def _get_period_(self, key):
+        period = key // self.period_length
+        return period
+
     def __getitem__(self, key):
-        self.highest_index = max(self.highest_index, key)
-        return self.default
+        if self.period_values:
+            period = self._get_period_(key)
+            return self.period_values[period]
+        else:
+            self.highest_index = max(self.highest_index, key)
+            return self.default
 
     def __init_list(self):
-        self.data = [self.default] * (self.highest_index + 1)
+        if self.period_values:
+            self.data = [
+                value
+                for value in self.period_values
+                for _ in range(self.highest_index)
+            ]
```

Comment:
I am unsure if I understood your example from above correct. As far as I see here if 
`s = _Sequence(period_values=[0,1,2], highest_index=9)` is passed we would have a list that is 3*9 = 27 long?

Reply:
Thanks for the hint, I missed correcting my first implementation. Now it's 9 time index in total, 3 per period.

And yes, you are right, the length of the years have to be the same. No mix of normal and leap year is possible.